crates/block-producer/src/store/mod.rs

```diff
@@ -163,6 +152,20 @@ impl DefaultStore {
     pub fn new(store: store_client::ApiClient<Channel>) -> Self {
         Self { store }
     }
+
+    /// Returns the latest block's header from the store.
+    pub async fn latest_header(&self) -> Result<BlockHeader, String> {
+        // TODO: fixup the errors types.
+        let response = self
```

Comment:
What is wrong with error types here? I'd either fix it or make the `TODO` more specific.

Reply:
I think the various store methods/trait could likely just use a single error enum instead of multiple for each. I've updated the comment to match this.